dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -82,22 +83,49 @@ public EventResource(EventDAO eventDAO, UserDAO userDAO) {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, notes = DESCRIPTION, responseContainer = "List", response = Event.class)
     public List<Event> getEvents(@Parameter(hidden = true) @ApiParam(hidden = true) @Auth User user, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
         User userWithSession = this.userDAO.findById(user.getId());
+        return getEventsForUser(userWithSession, eventSearchType, limit, offset);
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/{userId}")
+    @Operation(description = "No authentication.", summary = "Get events based on filter and user id.")
+    @ApiOperation(value = "List recent events for a user.", notes = "No authentication.", response = Event.class, responseContainer = "List")
+    public List<Event> getUserEvents(@ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
```

Comment:
If this method can yield more than one type of error response code (for example, a different response code for "user not found" versus another type of failure), it would be good to document each via annotations that propagate to `openapi.yaml` etc.  for example, see https://github.com/dockstore/dockstore/blob/7f536dd1572fff3b528c0b89a1dda8f65c37c0b8/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java#L226

Reply:
Good call, added the `SC_OK` response to both API calls https://github.com/dockstore/dockstore/pull/4901/commits/f2320d18907cbeadede10d1bf49873086113bfca and added `SC_NOT_FOUND` to the new endpoint https://github.com/dockstore/dockstore/pull/4901/commits/c6bbd7e689fc1e33f8db10802cd406870ab042cc. I don't believe there are any other response codes to handle for this request (let me know if anyone thinks of more).